zip-0320.rst

```diff
@@ -5,7 +5,242 @@
+   byte array (the **validating key hash**) should have a length of 20 bytes.
+3. Construct a new Unified Address using a single Traceable Receiver
+   :math:`\mathtt{0x04}` with the 20-byte **validating_key_hash** as
+   its value. In addition, metadata items such as an Address Expiry Height
+   or Address Expiry Date [#zip-0316-address-expiry]_ may be included.
+
+Wallets sending to a Traceable Unified Address MUST ensure that only
+transparent UTXOs are spent in the creation of a transaction.
+
+Reference Implementation (Alternative 2)
+----------------------------------------
+
+Javascript library `zcash_address_wasm`:
+
+.. code-block: javascript
+    import { to_traceable_address } from 'zcash_address_wasm'
+
+    // Create a deposit address that is valid for 30 days
+    var expiry_time = new Date();
+    expiry_time.setDate(expiry_time.getDate() + 30);
+
+    var traceable_addr = to_traceable_address('t1VmmGiyjVNeCjxDZzg7vZmd99WyzVby9yC', expiry_time.getTime() / 1000)
+
+Rust:
+   
+.. code-block: rust
+    use zcash_address::{
+        unified::{self, Encoding},
+        Network, ToAddress, TryFromAddress, ZcashAddress,
+    };
+
+    struct TraceableReceiver {
+        net: Network,
+        data: [u8; 20],
+    }
+
+    impl TraceableReceiver {
+        fn to_address(&self, expiry_time: u64) -> ZcashAddress {
+            let traceable_addr = unified::Address::try_from_items(vec![
+                unified::Receiver::Unknown {
+                    typecode: 0x04,
+                    data: self.data.to_vec(),
+                },
+                unified::Receiver::Unknown {
+                    typecode: 0xE0,
+                    data: expiry_time.to_le_bytes().to_vec(),
+                },
+            ])
+            .expect("We know that this produces a valid address.");
+
+            ZcashAddress::from_unified(self.net, traceable_addr)
+        }
+    }
+
+    impl TryFromAddress for TraceableReceiver {
+        type Error = unified::ParseError;
+
+        fn try_from_transparent_p2pkh(
+            net: Network,
+            data: [u8; 20],
+        ) -> Result<Self, zcash_address::ConversionError<Self::Error>> {
+            Ok(TraceableReceiver { net, data })
+        }
+    }
+
+.. [#BCP14] `Information on BCP 14 — "RFC 2119: Key words for use in RFCs to Indicate Requirement Levels" and "RFC 8174: Ambiguity of Uppercase vs Lowercase in RFC 2119 Key Words" <https://www.rfc-editor.org/info/bcp14>`_
+.. [#zip-0316-unified-addresses] `ZIP 316: Unified Addresses <zip-0316#encoding-of-unified-addresses>`_
+.. [#zip-0316-unified-requirements] `ZIP 316: Requirements for both Unified Addresses and Unified Viewing Keys <zip-0316#requirements-for-both-unified-addresses-and-unified-viewing-keys>`_
+.. [#zip-0316-address-expiry] `ZIP 316: <zip-0316#address-expiration-metadata>`_
+.. [#protocol-taddr] `Zcash Protocol Specification, Version 2023.4.0. Section 5.6.1.1 Transparent Addresses <protocol/protocol.pdf#transparentaddrencoding>`_
```

Comment:
```suggestion
.. [#protocol-transparentaddrencoding] `Zcash Protocol Specification, Version 2023.4.0. Section 5.6.1.1 Transparent Addresses <protocol/protocol.pdf#transparentaddrencoding>`_
```
(We consistently follow this convention in ZIPs of naming references to a protocol spec section as "protocol-" followed by the same anchor name as in the spec itself.)